src/clib/lib/enkf/obs_vector.cpp

```diff
@@ -81,29 +99,22 @@ __conf_instance_get_restart_nr(const conf_instance_type *conf_instance,
             conf_instance_get_item_value_double(conf_instance, "HOURS");
         util_inplace_forward_seconds_utc(&obs_time, hours * 3600);
     } else
-        util_abort("%s: Internal error. Invalid conf_instance?\n", __func__);
-
-    auto it = std::lower_bound(time_map.begin(), time_map.end(), obs_time);
-    const time_t tolerance = 30;
-    if (it == time_map.end() || std::abs(obs_time - *it) >= tolerance) {
+        throw std::logic_error(fmt::format(
+            "Observation: {} missing DATE, DAYS or HOURS", obs_key));
```

Comment:
`logic_error` isn't in `except.hpp` but can be added (one-line addition to that file)